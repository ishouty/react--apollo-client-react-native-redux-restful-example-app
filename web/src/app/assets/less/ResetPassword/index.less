.reset-password-container {

  font-family: @font;
  display: flex;
  flex-direction: column;
  justify-content: space-between;
  height: 100%;
  align-items: center;
  text-align: center;

  .reset-password-check-mail,
  .reset-password-main {
    max-width: 456px;

    .row {
      margin-top: 10px;
      text-align: center;
    }

    .spacing {
      margin-bottom: 10px;
    }

  }

  .firstMessage b {
    color: @color-dark-gray;
  }

  a {
    font-family: @font;
  }

  .header-bar a {
    padding-left: 15px;
  }

  .loading-btn-container {
    left: 47%;
  }

  h1 {
    font-size: 40px;
    text-align: center;
    margin-bottom: 20px;
    font-family: @fontLight;

  }

  h5 {
    font-size: 16px;
    font-family: @font;
    color: @color-gray;
    margin-bottom: 30px;
  }

  .main-container {
    padding: 0px 25%;
  }

  .send-container {
    text-align: center;
  }

  .reset-container {
    margin-top: 20px;
  }

}