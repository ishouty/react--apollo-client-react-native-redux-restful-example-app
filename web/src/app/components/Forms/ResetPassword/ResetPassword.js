import React from "react"
import {Field, reduxForm} from "redux-form"
import {resetPassword} from "../../../actions/auth"
import {browserHistory, Link} from "react-router"
import {ResetPasswordText, LoginText, CommonText} from "../../../constants/text/text"
import {maxLength} from "../../../utils/HelperUtils"
import {Button, FormGroup, FormControl, HelpBlock, Row, Col} from "react-bootstrap"

const validate = values => {

    const errors = {}
    const requiredFields = ["email"]

    requiredFields.forEach(field => {
        if (!values[field]) {
            errors[field] = "Required"
        }
    })

    if (values.email && !/^(([^<>()[\]\\.,;:\s@\"]+(\.[^<>()[\]\\.,;:\s@\"]+)*)|(\".+\"))@((\[[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\])|(([a-zA-Z\-0-9]+\.)+[a-zA-Z]{2,}))$/.test(values.email)) {
        errors.email = LoginText.invalidEmailAddress
    }
    return errors

}

const renderFieldText = ({id, input, label, type, disabled, placeholder, helpBlock, meta: {touched, error, warning}}) => (

    <div>
        {
            disabled == "true" ? <FormControl {...input} id={id} type={type} placeholder={placeholder} disabled/>
                :
                <FormControl {...input} id={id} type={type} placeholder={placeholder}/>
        }

        {
            (touched && ((error && <span className="error-block">{error}</span>) || (warning &&
                <span className="warning-block">{warning}</span>)) || <HelpBlock>{helpBlock}</HelpBlock>)
        }
    </div>

)

const getStyles = () => {

    return {
        form: {},
        helpBlock: {}
    }

}

const ResetPasswordForm = props => {

    const {error, handleSubmit, submitting} = props

    const styles = getStyles()

    return (

        <div>

            <form onSubmit={handleSubmit(resetPassword)}>

                <FormGroup bsSize="large">

                    <Field id="email" type="email" name="email"
                           placeholder={ResetPasswordText.placeholder} validate={[maxLength(254)]}
                           component={renderFieldText}/>

                    <HelpBlock style={styles.helpBlock}></HelpBlock>

                </FormGroup>

                <div>
                    {error && <strong>{error}</strong>}
                </div>

                <Row className="send-container no-margin">
                    <Button id="submit_btn" type="submit" bsSize="large"
                            onClick={() => {
                                handleSubmit(resetPassword)
                            }}
                            className="black-btn" disabled={submitting}>{CommonText.sendButton}</Button>
                </Row>

                <Col>
                    <Row>
                        <Link to="/">{CommonText.loginButton}</Link>
                    </Row>
                </Col>
            </form>

        </div>

    )

}

export default reduxForm({
    form: "ResetPasswordForm",  // a unique identifier for this form
    validate,
    resetPassword
})(ResetPasswordForm)
