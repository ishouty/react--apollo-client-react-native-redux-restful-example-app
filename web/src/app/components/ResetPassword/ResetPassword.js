import React, { Component } from 'react'
import { connect } from 'react-redux'
import { browserHistory } from 'react-router'
import { Row } from 'react-bootstrap'
import { LoadingButton } from '../Common/Common'
import * as actionTypes from '../../actions/actionTypes'
import {
  ResetPasswordText,
  ResetPasswordCheckEmailText
} from '../../constants/text/text'
import ResetPasswordForm from '../Forms/ResetPassword/ResetPassword'
import Footer from '../Layout/Footer/Footer'
import PropTypes from 'prop-types'

@connect((store) => {
  return {
    resetPassword: store.users,
    profile: store.profile
  }
})
class ResetPassword extends Component {
  componentWillMount() {
    browserHistory.listen(() => {
      this.resetState(this.props.dispatch)
    })
  }

  resetState() {
    this.props.dispatch({
      type: actionTypes.RESET_PASSWORD,
      payload: { completeForm: false, email: null }
    })
  }

  BackToResetPassword(dispatch) {
    dispatch({
      type: actionTypes.RESET_PASSWORD,
      payload: { completeForm: false, email: null }
    })
  }

  render() {
    const { resetPassword, dispatch } = this.props

    return (
      <div className="reset-password-container">
        <div></div>
        <div>
          <LoadingButton />

          {resetPassword.completeResetPassword.completeForm ==
          false ? (
            <Row className="reset-password-main">
              <Row>
                <h1>{ResetPasswordText.title}</h1>
                <h5>{ResetPasswordText.description}</h5>
              </Row>

              <ResetPasswordForm dispatch={dispatch} />
            </Row>
          ) : (
            <Row className="reset-password-check-mail">
              <Row style={{ textAlign: 'center' }}>
                <h1>{ResetPasswordCheckEmailText.title}</h1>

                <Row className="firstMessage">
                  <span>
                    {ResetPasswordCheckEmailText.description[0]}
                    <b>{resetPassword.completeResetPassword.email}</b>
                  </span>
                </Row>

                <Row className="spacing">
                  <span>
                    {ResetPasswordCheckEmailText.description[1]}
                  </span>
                </Row>

                <Row>
                  <span>
                    {ResetPasswordCheckEmailText.description[2]}
                  </span>
                </Row>

                <Row className="spacing">
                  <span>
                    {ResetPasswordCheckEmailText.description[3]}
                  </span>
                </Row>

                <Row className="spacing">
                  <a
                    onClick={() => {
                      this.BackToResetPassword(dispatch)
                    }}
                  >
                    {ResetPasswordCheckEmailText.link}{' '}
                  </a>
                </Row>
              </Row>
            </Row>
          )}
        </div>
        <Footer />
      </div>
    )
  }
}

ResetPassword.propTypes = {
  dispatch: PropTypes.func,
  resetPassword: PropTypes.object
}

export default ResetPassword
