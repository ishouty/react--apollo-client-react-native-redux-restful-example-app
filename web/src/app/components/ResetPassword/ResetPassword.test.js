import React, {Component} from 'React'
import {shallow, mount } from 'enzyme'
import toJson from 'enzyme-to-json'
import ResetPassword from './ResetPassword'


describe('ResetPassword', ()=> {

    test('Component should render and display',()=> {

        const props = {
            resetPassword: {
                completeResetPassword: {
                    completeForm: false
                }
            }
        }

        const component = shallow(<ResetPassword {...props} />)
        const tree = toJson(component)
        expect(tree).toMatchSnapshot()
    })

})

