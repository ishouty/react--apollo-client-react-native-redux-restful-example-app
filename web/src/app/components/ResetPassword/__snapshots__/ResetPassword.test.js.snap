// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ResetPassword Component should render and display 1`] = `
<ContextConsumer>
  <Component />
</ContextConsumer>
`;
